src/aead.rs

```diff
@@ -148,13 +149,179 @@ pub fn open(
+///
+/// let key = SecretKey::default();
+///	let mut sealer = StreamSealer::new(&key).unwrap();
+///	let nonce = sealer.get_nonce();
+/// let mut opener = StreamOpener::new(&key, &nonce).unwrap();
+///
+/// //Message 1
+///	let plaintext1 = "Secret message 1".as_bytes().to_vec();
+/// let cipher1 = sealer
+///		.seal_chunk(&plaintext1, None, StreamTag::MESSAGE)
+///		.unwrap();
+/// let (dec1, tag1) = opener.open_chunk(&cipher1, None).unwrap();
+/// assert_eq!(plaintext1, dec1);
+/// assert_eq!(tag1, StreamTag::MESSAGE);
+///
+/// //Message 2
+/// let plaintext2 = "Secret message 2".as_bytes().to_vec();
+/// let cipher2 = sealer
+///		.seal_chunk(&plaintext2, None, StreamTag::MESSAGE)
+///		.unwrap();
+/// let (dec2, tag2) = opener.open_chunk(&cipher2, None).unwrap();
+/// assert_eq!(plaintext2, dec2);
+/// assert_eq!(tag2, StreamTag::MESSAGE);
+///
+/// //Message 3 (Last message of this stream, using the FINISH tag)
+/// let plaintext3 = "Secret message 3".as_bytes().to_vec();
+/// let cipher3 = sealer
+///		.seal_chunk(&plaintext3, None, StreamTag::FINISH)
+///		.unwrap();
+/// let (dec3, tag3) = opener.open_chunk(&cipher3, None).unwrap();
+/// assert_eq!(plaintext3, dec3);
+/// assert_eq!(tag3, StreamTag::FINISH);
+///
+/// # Ok::<(), orion::errors::UnknownCryptoError>(())
+/// ```
+pub struct StreamSealer {
+	internal_sealer: aead::streaming::StreamXChaCha20Poly1305,
+	nonce: Nonce,
+}
+
+impl StreamSealer {
+	/// Returns the [`Nonce`](../hazardous/stream/xchacha20/struct.Nonce.html) used.
+	/// The nonce does not need to be kept secret
+	pub fn get_nonce(&self) -> Nonce {
+		self.nonce
+	}
+
+	#[must_use = "SECURITY WARNING: Ignoring a Result can have real security implications."]
+	/// Initialize a `StreamSealer` struct with a given key.
+	pub fn new(secret_key: &SecretKey) -> Result<Self, UnknownCryptoError> {
+		let nonce = Nonce::generate();
+		let sk = &aead::streaming::SecretKey::from_slice(secret_key.unprotected_as_bytes())?;
+
+		Ok(Self {
+			internal_sealer: aead::streaming::StreamXChaCha20Poly1305::new(&sk, &nonce),
+			nonce,
+		})
+	}
+
+	#[must_use = "SECURITY WARNING: Ignoring a Result can have real security implications."]
+	/// Encrypts `plaintext` and additionally authenticates `ad`. The [`StreamTag`](../hazardous/aead/streaming/enum.StreamTag.html)
+	/// indicates the type of message.
+	pub fn seal_chunk(
+		&mut self,
+		plaintext: &[u8],
+		ad: Option<&[u8]>,
+		tag: StreamTag,
+	) -> Result<Vec<u8>, UnknownCryptoError> {
+		let sealed_chunk_len = plaintext.len().checked_add(aead::streaming::ABYTES);
+		if sealed_chunk_len.is_none() {
+			return Err(UnknownCryptoError);
+		}
+
+		let mut sealed_chunk = vec![0u8; sealed_chunk_len.unwrap()];
+		self.internal_sealer
+			.seal_chunk(plaintext, ad, &mut sealed_chunk, tag)?;
+
+		Ok(sealed_chunk)
+	}
+}
+
+/// TODO: Link does not work on index site
+/// Decrypts messages encrypted by a `StreamSealer`.
+/// See [`StreamSealer`](struct.StreamSealer.html) for documentation
```

Comment:
This link works fine on the `aead` index site for me. Any specific issue you encountered?